File: core/common/src/main/java/com/shadov/codehellven/common/model/Languages.java
package com.shadov.codehellven.common.model;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum Languages {
	JAVA, SCALA;

	@JsonCreator
	public static Languages fromString(String key) {
		for (Languages type : Languages.values()) {
			if (type.name().equalsIgnoreCase(key))
				return type;
		}
		return null;
	}
}

File: core/common/src/main/kotlin/com/shadov/codehellven/common/model/Languages.kt
package com.shadov.codehellven.common.model

import com.fasterxml.jackson.annotation.JsonCreator

enum class Languages {
    JAVA, SCALA;

    companion object {
        @JsonCreator
        fun fromString(key: String): Languages {
            for (type in Languages.values()) {
                if (type.name.equals(key, ignoreCase = true))
                    return type
            }
            throw IllegalArgumentException("Language with key $key does not exist")
        }
    }
}
